cmd/handler-utils.go

```diff
@@ -39,7 +39,7 @@ func parseLocationConstraint(r *http.Request) (location string, s3Error APIError
 	// be created at default region.
 	locationConstraint := createBucketLocationConfiguration{}
 	err := xmlDecoder(r.Body, &locationConstraint, r.ContentLength)
-	if err != nil && err != io.EOF {
+	if err != nil && r.ContentLength != 0 {
```

Comment:
This is not correct, location constraint is optional. The previous code is correct

Reply:
Yeah. The idea was to check if it's bigger than zero, since that should cover the case when it's not passed. must have missed that somewhere in the process